### PR TITLE
Yarn completion tweaks

### PR DESCRIPTION
**Pull request by nthapaliya**
## Description

Like the npm completion script, parses package.json (if available) with jq (if available) to provide completions for `yarn run`

Fixes issue #4597

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

### CODE REVIEW

File: share/completions/npm.fish [hunk 1]
```diff
@@ -70,22 +70,14 @@ end
...
 function __fish_npm_run
     # Like above, only try to call npm if there's a command by that name to facilitate aliases that call nvm.
     if command -sq jq; and test -e package.json
-        jq -r '.scripts | to_entries[] | .key,.value' <package.json | __fish_parse_npm_run_completions
+        jq -r '.scripts | to_entries | map("\(.key)\t\(.value | tostring | .[0:20])") | .[]' package.json
+    else if command -sq npm; and command -sq jq
+        command npm run --json | jq -r 'to_entries | map("\(.key)\t\(.value | tostring | .[0:20])") | .[]'
     else if command -sq npm
-        command npm run | string match -r -v '^[^ ]|^$' | string trim | __fish_parse_npm_run_completions
+        command npm run | string match -r -v '^[^ ]|^$' | string trim | awk 'NR % 2 == 1 { name=$0 ; next } { print name "\t" substr($0, 0, 20) }'
```
Comment (faho) [inline share/completions/npm.fish]: The function has the advantage that it uses only builtins. With awk, it's a bit of a crapshoot which things are supported on which implementation, so we try not to rely on it unless it can't be helped.

Is there a specific reason to change this other than that it's shorter?

Comment (nthapaliya) [reply share/completions/npm.fish]: Thats a good point about different awk implementations having different behaviors. I only tested this on my mac, (`GNU Awk 4.2.0, API: 2.0 (GNU MPFR 3.1.6, GNU MP 6.1.2)
`), so I don't really know if this will break elsewhere.

There was no specific reason other than it's shorter and avoided defining another function. I can revert the changes to this file if you want.

Comment (faho) [reply share/completions/npm.fish]: >I can revert the changes to this file if you want.

No need, it's in a different commit already. I can just not merge that one.

**Comment by faho** [comment]
In case anyone is wondering on how to merge this with git, it's actually quite simple:

- Check out the PR (e.g. with https://gist.github.com/piscisaureus/3342247, you can just `git fetch --all; git checkout pr/4601`)

- Run `git rebase --interactive master`, which will open your text editor.

- Comment out the commit you don't want (with a `#`). Here you can also reorder commits or squash them together or do a bunch of other stuff.

- Save changes and exit.

- `git checkout master; git merge pr/4601`

- Do a `git log` to check for any mistakes

- `git push origin`

And that way you end up with 7cd9c3b2bf8eebebebee02bd1de67a43938401bc..0c2d939e02779a80561d2e77aad3a4056bb230b8.

TL;DR: Thanks, merged!
